File: Python/ex12.py
# 12. Create simple calculator

def calculator():
    first_digit = int(input("Type a digit: "))
    outcome = first_digit
    prompt_text = "Type another digit: "
    while True:
        choice = str(input("Type an operator (+, -, *, /, =): "))
        choice_trimmed = choice.strip()
        is_digit = choice.isdigit()
        if is_digit:
            print("You typed a digit, please type an operator (+, -, *, /, =): ")
        else:
            match choice_trimmed:
                case "+":
                    digit = int(input(prompt_text))
                    outcome = outcome + digit
                case "-":
                    digit = int(input(prompt_text))
                    outcome = outcome - digit
                case "*":
                    digit = int(input(prompt_text))
                    outcome = outcome * digit
                case "/":
                    digit = int(input(prompt_text))
                    outcome = outcome / digit
                case "=":
                    return outcome

print("Outcome =", calculator())

File: JavaScript/ex12.js
// 12. Create simple calculator

const prompt = require ("prompt-sync")();

function digit(){
    let number = prompt("Type a digit: ", 10);
    while (Number.isInteger(+number) !== true){
        number = prompt("Wrong digit, please type a correct digit: ", 10);
    }
    return parseInt(number, 10);
}

function calculator(){
    let outcome = digit();
    while (true){
        let choice = String(prompt("Type an operator (+, -, *, /, =): "));
        switch (choice.trim()){
            case ("+"):
                outcome = outcome + digit();
                break;
            case ("-"):
                outcome = outcome - digit();
                break;
            case ("*"):
                outcome = outcome * digit();
                break;
            case ("/"):
                outcome = outcome / digit();
                break;
            case ("="):
                return outcome;
            default:
                console.log("Wrong operator")
                break;
        }
    }
}

console.log("Outcome =", calculator())